integration/tests/neon_evm/test_simulate_solana.py

```diff
@@ -0,0 +1,695 @@
+    ):
+        # Create Neon transaction
+        neon_signed_tx = eth_utils.make_contract_call_trx(
+            evm_loader=evm_loader,
+            user=sender_with_tokens,
+            contract=calculator_contract,
+            function_signature="callCalculator()",
+        )
+        simulated_compute_units = actual_compute_units = index = 0
+        done = done_simulation = done_execution = False
+
+        while not done and index <= 500:
+            # Create a Solana transaction
+            sol_tx = instructions.TransactionWithComputeBudget(operator_keypair)
+            operator_balance_pubkey = evm_loader.get_operator_balance_pubkey(operator_keypair)
+            sol_tx.add(
+                instructions.make_PartialCallOrContinueFromRawEthereumTX(
+                    index=index,
+                    step_count=500,
+                    instruction=neon_signed_tx.rawTransaction,
+                    operator=operator_keypair,
+                    operator_balance=operator_balance_pubkey,
+                    evm_loader_id=evm_loader.loader_id,
+                    storage_address=holder_acc,
+                    treasury=treasury_pool,
+                    additional_accounts=[
+                        sender_with_tokens.balance_account_address,
+                        calculator_contract.solana_address,
+                        calculator_contract.solana_address,
+                    ],
+                )
+            )
+            sol_tx.sign(operator_keypair)
+
+            # Simulate the transaction
+            if not done_simulation:
+                assert not done_execution, f"Execution completed in {index} steps but simulation is still going"
+
+                serialized_transaction = sol_tx.serialize()
+                hex_serialized_transaction = serialized_transaction.hex()
+                simulate_response = neon_api_client.simulate_solana(
+                    blockhash=os.urandom(32).hex(),
+                    transactions=[hex_serialized_transaction],
+                )
+                simulated_transactions = simulate_response.json()["value"]["transactions"]
+
+                simulated_compute_units += sum(
+                    [simulate_result["executed_units"] for simulate_result in simulated_transactions]
+                )
+
+                for simulated_transaction in simulated_transactions:
+                    if simulated_transaction["error"]:
+                        raise AssertionError(f"Error in sol trx: {simulated_transaction}")
+
+                    for log in simulated_transaction["logs"]:
+                        if "ExitError" in log:
+                            raise AssertionError(f"EVM Return error in logs: {simulated_transaction}")
+
+                        elif "exit_status" in log:
+                            done_simulation = True
+                            break
+
+            # Execute the transaction
+            if not done_execution:
```

Comment:
You have a lot of the same code in each test
If it is possible please move something in functions (for example this check)